packages/backend/src/peripherals/starkware/L2TransactionDownloader.ts

```diff
@@ -0,0 +1,56 @@
+import { Logger } from '../../tools/Logger'
+import { L2TransactionRepository } from '../database/L2TransactionRepository'
+import { StateUpdateRecord } from '../database/StateUpdateRepository'
+import { FeederGatewayClient } from './FeederGatewayClient'
+
+export class L2TransactionDownloader {
+  constructor(
+    private readonly feederGatewayClient: FeederGatewayClient,
+    private readonly transactionRepository: L2TransactionRepository,
+    private readonly logger: Logger
+  ) {}
+
+  async sync(
+    stateUpdates: Pick<StateUpdateRecord, 'id' | 'batchId' | 'blockNumber'>[]
+  ) {
+    const latestSyncedTransactionStateUpdateId =
+      (await this.transactionRepository.findLatestStateUpdateId()) ?? 0
+
+    for (const stateUpdate of stateUpdates) {
```

Comment:
We should start syncing from `latestSyncedTransactionStateUpdateId` which can be 0. We should not just loop stateUpdates. I don't think we need a list of stateUpdates, just the latest id....